core/src/main/java/org/elasticsearch/rest/action/search/RestSearchAction.java

```diff
@@ -79,22 +80,27 @@ public RestSearchAction(Settings settings, RestController controller, Client cli
 
     @Override
     public void handleRequest(final RestRequest request, final RestChannel channel, final Client client) {
-        SearchRequest searchRequest;
-        searchRequest = RestSearchAction.parseSearchRequest(request, parseFieldMatcher);
+        SearchRequest searchRequest = new SearchRequest();
+        RestSearchAction.parseSearchRequest(searchRequest, request, parseFieldMatcher, null);
         client.search(searchRequest, new RestStatusToXContentListener<SearchResponse>(channel));
     }
 
-    public static SearchRequest parseSearchRequest(RestRequest request, ParseFieldMatcher parseFieldMatcher) {
-        String[] indices = Strings.splitStringByCommaToArray(request.param("index"));
-        SearchRequest searchRequest = new SearchRequest(indices);
+    public static void parseSearchRequest(SearchRequest searchRequest, RestRequest request, ParseFieldMatcher parseFieldMatcher,
```

Comment:
This diverges a bit from how master works because most of the parsing can't be done on the coordinating node. So this change is smaller than master. Here we just change this API so that it can modify a SearchRequest rather than create one. This lets reindex create a SearchRequest, change the defaults on it to be reindex's defaults, and then pass the search request through this API so that REST parameters still work in the same way. Mostly. Reindex does quite a bit of hacking on the body to transform bits of it's API into one that this search request likes. We'll get to that later.